src/mongo/delegates/Views.class.php

```diff
@@ -599,8 +600,37 @@ protected function doJoins($source, $joins, &$dest, $from, $contextAlias, $build
                     {
                         // make sure any sequences are expanded before extracting properties
                         if (isset($ruleset['joins'])) $this->expandSequence($ruleset['joins'],$linkMatch);
-
-                        $dest[_GRAPHS][] = $this->extractProperties($linkMatch,$ruleset,$from);
+                        if (isset($ruleset['filter']))
+                        {
+                            foreach($ruleset['filter'] as $filterPredicate => $filter)
+                            {
+                                foreach($filter as $filterType => $filterMatch)
+                                {
+                                    if (isset($linkMatch[$filterPredicate]))
```

Comment:
We're agreeing that $linkMatch is okay then. 

The $filterPredicate on this line is coming from the view. So what your saying is that someone could have specified the fquri for the filter predicate in the view? Something like:

```javascript
{
    "stores": {
        "some_store": {
            "pods" : {},
            "view_specifications": [
                {
                    "_id": "v_spec_1",
                    "from": "CBD_something",
                    "type": "owl:Thing"
                    "joins": {
                        "dct:isVersionOf": {
                            "filter" : {
                                "http://xmlns.com/foaf/0.1/name" : {"l" : "John"}
                            },
                            "include": ["dct:subject", "rdf:type"]
                        }
                    }
                }
            ]
        }
    }
}
```

i.e. the filter is using "http://xmlns.com/foaf/0.1/name" not "foaf:name".

Again - this is easy to do - we have a labeller in the view - but is this really supported for all the other parts of the view? Do we really want to support that here? Or am I misunderstanding what your saying?